src/tests/xkey/test13.vtc

```diff
@@ -0,0 +1,118 @@
+varnishtest "Make many requests and purge them all"
+
+server s1 {
```

Comment:
I think you can get away with:

```
server s1 -repeat 11 {
  rxreq
  rxresp
}
```

(same thing for `c1`) and in the vcl:

```
sub vcl_backend_response {
  set beresp.http.xkey = bereq.xid;
}
```

Reply:
We need to fill up the xkey_pool, which requires deleting more than POOL_MAX xkeys. However...now that I think about it I don't think we need to fill up the pool. We just need one item to be in the pool to cause the bug we caused before.

1. Cache an item.
2. Purge it.
3. Cache another item, whose memory is taken from the pool.
4. Purge again.

The g_oc_bytes gauge would already underflow at this point. Our test would fail after step 3 because it would expect the gauge to increase and in the pre-patched version of the code it didn't do that.

Reply:
That worked. We simplified the test and it still is able to catch our mistake. 👍